shuup/admin/modules/suppliers/views/delete.py

```diff
@@ -0,0 +1,38 @@
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2019, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.conf import settings
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import RedirectView
+
+from shuup.core.models import Supplier
+
+
+class SupplierDeleteView(RedirectView):
```

Comment:
Why his is redirect view? Could this be same as in [here](https://github.com/shuup/shuup/blob/master/shuup/admin/modules/products/views/delete.py).

```
class SupplierDeleteView(DetailView):
    model = Supplier
    context_object_name = "supplier"

    def get_success_url(self):
        return reverse("shuup_admin:supplier.list")

    def get_queryset(self, *args, **kwargs):
        suppier = self.get_object()
        shop = get_shop(self.request)
        qs = Supplier.objects.filter(shops=shop)
        admin_supplier = get_supplier(self.request)
        if admin_supplier:
            # If admin has supplier enabled allow only delete self
            # althought not good view to enable for vendor
            qs = qs.filter(id=admin_supplier.pk)
        return qs

    def get(self, request, *args, **kwargs):
        supplier = self.get_object()
        return HttpResponseRedirect(get_model_url(supplier, shop=self.request.shop))

    def post(self, request, *args, **kwargs):
        supplier = self.get_object()
        supplier.soft_delete(user=request.user)
        messages.success(request, _(u"%s has been marked deleted.") % supplier)
        return HttpResponseRedirect(self.get_success_url())



class MultivendorSupplierDeleteView(SupplierDeleteView):
    def get_success_url(self):
        return reverse("shuup_admin:shuup_multivendor.vendor.list")
```

Reply:
`DetailView` used to render detailed template views, in our case, SupplierDeleteView result is redirect, so my choice was `RedirectView`

Reply:
This is not huge thing as long as it works. You just need to make sure that current user can only delete suppliers that is linked to current shop and supplier active in admin (see the get_queryset).